src/test/java/com/cdancy/bitbucket/rest/features/BranchApiMockTest.java

```diff
@@ -100,6 +114,33 @@ public void testListBranches() throws Exception {
         }
     }
 
+
+    private static class BuildStatusInfo {
```

Comment:
This is too specific IMO and won't work across the board. I think the correct approach here would be to use `JsonParser` which is already on the classpath. Something like this might work:

```
    public abstract Map<String, JsonObject> metadata();

    Branch() {
    }

    @SerializedNames({ "id", "displayId", "type", "latestCommit", "latestChangeset", "isDefault", "errors", "metadata" })
    public static Branch create(String id, String displayId, String type,
                                String latestCommit, String latestChangeset, boolean isDefault, List<Error> errors, Map<String, Object> metadata) {
        JsonParser parser = new JsonParser(); // could probably just use the static instance from our fallbacks class
        Map<String, JsonObject> convertedMetadata = Maps.newHashMap();
        if (metadata != null) {
             metadata.each { // using groovy/pseudo-code here
                  convertedMetadata.put(key, parser.parse(value.toString()).getAsJsonObject();
             }
        }

        return new AutoValue_Branch(Utils.nullToEmpty(errors), id, displayId, type, latestCommit, latestChangeset, isDefault, convertedMetadata);
    }
```